presto-spi/src/main/java/com/facebook/presto/spi/function/SqlInvokedFunction.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.function;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface SqlInvokedFunction
```

Comment:
What are the other potential kind of `SqlInvokedFunction` other than `SqlInvokedRegularFunction`?

Reply:
Convert `SqlInvokedFunction`, `SqlFunctionId`, and `SqlFunctionHandle` to concrete classes.